opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1056,25 +1069,31 @@ namespace Opm {
                         const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
                         const auto& fs = intQuants.fluidState();
                         const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
-                        hcpv[region] += pv[c] * hydrocarbon;
-                        pres[region] += pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value();
+                        const double pv =
+                            ebosSimulator_.model().dofTotalVolume(c)
+                            * ebosSimulator_.problem().porosity(c);
+                        hcpv[region] += pv * hydrocarbon;
+                        pres[region] += pv * fs.pressure(FluidSystem::oilPhaseIdx).value();
                     }
                 }
                 for (int c = 0; c < nc; ++c) {
```

Comment:
Here you still use the old loop style and the obfuscated one from above. Why? Seems inconsistent and will raise questions in the future.